### DISCUSSION TITLE
morph tools/export

### DISCUSSION THREAD

**Discussion by harbuzenger** [soupday/cc3_blender_tools#71] [Q&A] [answered]
Hi All,

I've just followed a tutorial where the guy exports some morph targets for use in CC3.  That was in version 1.0.2.  In 1.3.3 I can't find this button.  Is there a new way to do this?

Many thanks

Gaz

**Comment by soupday** [comment, marked as answer]
Just use the "Export To CC3" button.

It will export the character in the same format as it was imported. So if you import an OBJ (nude character in bind pose) it will export it back in the same format for making morph targets.

**Reply by harbuzenger** [reply]
If I start with Blender and an .obj of the mesh, the export to CC3 is greyed out.  I can't import the .obj directly into CC3 because it needs the objkey file...

**Comment by soupday** [comment]
I think I will change the name of the button to "Export Morph Target" if the import was an OBJ (and has an .objkey file.)
Should be less confusing.

**Reply by harbuzenger** [reply]
Hey, thanks for the reply.  It's that bit that's confusing.

I'm starting from a 3D scan you see.  So I've used a mix of zbrush and wrap3d to get so far, but it's all quite painful.  So I'm trying to consolidate as much as I can with Blender.

When I bring in an .obj to Blender though, I can't export it to CC3 with the current button because it doesn't have the required vertex groups and armature deformer (although the topology is correct to what CC3 expects).  So I had hoped to just export a morph obj and use that to get the required body shape I want, then start from there.

Maybe there's a whole tutorial on this that I've missed?

Gaz

**Comment by soupday** [comment]
If you want a base character OBJ to reshape and bring back in as a morph, then you must export as "Nude character in bind pose.", otherwise no ObjKey and CC3 will not re-import it.

Also: no changes in topology or vertex order.

**Reply by soupday** [reply]
it's CC3 that generates the ObjKey with the initial export, that key is then used to get the OBJ mesh back into CC3. The ObjKey contains all the infomation needed to reconstruct the mesh as a character morph in CC3. It's not something I can generate myself.

But it only generates the ObjKey when exporting from CC3 in the bind pose.

**Reply by harbuzenger** [reply]
So yeah, I'm exporting as "Nude character in bind pose." from CC3.  I get the obj (it's massive but I scale it down and apply the scale).  I import that into Blender.  I apply a blendshape (shape key in Blender) (from the sculpt/wrap I did).  The blendshape works, the mesh looks correct.  I export from Blender as obj and overwrite the one I exported from CC3 at the beginning of the process. 

In CC3 I go through the "create morph" process, and select the obj file and it auto completes the objkey field.  Great.  

Then it errors and complains that the objkey is corrupt.

I'm clearly not doing something correctly in the export/import process. 

If I start the process in Blender by importing the obj CC3 tools always has a greyed out export button.  So I'm going via the standard Blender menu...

Thanks for all the replies so far, I really appreciate it

**Reply by soupday** [reply]
Generally it's a good idea to understand the manual procedure anyway so you're on the right track.

You need some quite specific import and export settings to ensure that the vertex order is preserved:

Import settings: 
(Keep Vertex Order + Poly Groups)
Note: Scale and apply mesh by 0.01 after import
![i](https://user-images.githubusercontent.com/79094830/165840164-b73c00d2-2519-4ef8-b2b1-8d481dc2931e.png)

Export Settings:
(Only Selected, 100.0 Scale, No materials, Vertex Groups)
![e](https://user-images.githubusercontent.com/79094830/165841584-ef4004b7-e408-461d-9751-a0ee5f4dff73.png)

Note: these are Blender 3.1 settings. If you have an earlier version you may need "Keep Vertex Order"  for the export settings too.
CC3 is very particular about these settings.



About the greyed out export button, to rule out the obvious: You are importing the .obj via the "Import Character" button in the CC3 Tools panel?

**Reply by harbuzenger** [reply]
haha, holy ballbags, no I didn't do that.  Now I have done that, and I'm in business - still with tons of hassle (probably through my own doing).  I'll get there I'm sure.  Thank you!!!

I now at least have a working slider for the basic face shape that I want in CC3.  Just got the other 98% of the work to do now!

I fixed vertex order with this actually.  Very handy tool.

https://bartoszstyperek.gumroad.com/l/copy_verts_ids

Thanks again, I'm sure I'll be back
